client.go

```diff
@@ -1423,6 +1489,93 @@ func (c *Client[TTx]) insertManyParams(params []InsertManyParams) ([]*riverdrive
 	return insertParams, nil
 }
 
+func (c *Client[TTx]) maybeNotifyInsert(ctx context.Context, execTx riverdriver.ExecutorTx, state rivertype.JobState, queue string) error {
+	if state != rivertype.JobStateAvailable {
+		return nil
+	}
+	return c.maybeNotifyInsertForQueues(ctx, execTx, []string{queue})
+}
+
+func (c *Client[TTx]) maybeNotifyInsertMany(ctx context.Context, execTx riverdriver.ExecutorTx, insertParams []*riverdriver.JobInsertFastParams) error {
+	queues := sliceutil.Map(insertParams, func(params *riverdriver.JobInsertFastParams) string { return params.Queue })
+	return c.maybeNotifyInsertForQueues(ctx, execTx, queues)
```

Comment:
Excellent catch, thank you. I added a TODO for the moment because I want to make sure I get test coverage on this, will for sure fix that before merging.

Reply:
Added test coverage for notifying on `Available` only, both on `Insert` and `InsertMany` 🙌  Definitely a gap prior to this. Lmk if it looks alright!

Reply:
Awesome. Left one more comment about possible use of a map, but looks good.